gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/CaResources.java

```
@@ -53,113 +62,73 @@ public CaResources(ILogger logger) {
-                    .toString();
-            List<String> checkCertificateCommand = List.of(
-                    keytoolPath,
-                    "-list",
-                    "-storepass",
-                    "changeit",
-                    "-alias",
-                    alias,
-                    "-keystore",
-                    jdkInstallationDirectory.resolve("lib/security/cacerts").toString());
-            CommandRunner.runWithOutputCollection(new ProcessBuilder().command(checkCertificateCommand));
-            return true;
-        } catch (Exception e) {
-            if (e.getMessage().contains(String.format("Alias <%s> does not exist", alias))) {
-                return false;
+            LdapName ldapName = new LdapName(distinguishedName);
+            for (Rdn rdn : ldapName.getRdns()) {
+                if ("CN".equalsIgnoreCase(rdn.getType())) {
+                    return String.format("GradleJdks_%s", ((String) rdn.getValue()).replaceAll("\\s", ""));
+                }
             }
-            throw new RuntimeException("Unable to check if the certificate already exists in the truststore", e);
+        } catch (InvalidNameException e) {
+            logger.logError(String.format("Failed to extract ldapName from %s", distinguishedName));
+        }
+        return String.format("GradleJdks_%s", distinguishedName.replaceAll("\\s", ""));
+    }
+
+    private Optional<KeyStore> loadKeystore(char[] password, Path location) {
+        try (InputStream keystoreStream = new BufferedInputStream(Files.newInputStream(location))) {
+            KeyStore keystore = KeyStore.getInstance("JKS");
+            keystore.load(keystoreStream, password);
+            return Optional.of(keystore);
+        } catch (KeyStoreException | CertificateException | NoSuchAlgorithmException | IOException e) {
+            logger.log(String.format("Couldn't load jks, an exception occurred %s", e));
+            return Optional.empty();
```

I'd just have this throw here rather than returning an optional, losing information, then rethrowing via `orElseThrow`. This way we can keep whole stacktrace.